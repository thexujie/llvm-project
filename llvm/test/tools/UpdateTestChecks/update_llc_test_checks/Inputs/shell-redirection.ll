;; Check that we can parse RUN lines that have shell redirections
; RUN: llc -mtriple=x86_64 %s -o - 2>/dev/null | FileCheck %s --check-prefix=I32
; RUN: llc < %s -mtriple=x86_64 2>&1 | FileCheck %s --check-prefix=I32
; RUN: sed 's/i32/i64/g' %s | llc -mtriple=x86_64 2>&1 | FileCheck %s --check-prefix=I64

define i32 @add(i32 %X, i32 %Y) {
  %Q = add i32 %X, %Y
  ret i32 %Q
}
