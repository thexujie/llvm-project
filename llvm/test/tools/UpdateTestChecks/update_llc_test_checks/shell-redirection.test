## Check that update_llc_test_checks can parse RUN lines with shell redirection
# RUN: cp -f %S/Inputs/shell-redirection.ll %t.ll && %update_llc_test_checks %t.ll
# RUN: diff -u %t.ll %S/Inputs/shell-redirection.ll.expected
