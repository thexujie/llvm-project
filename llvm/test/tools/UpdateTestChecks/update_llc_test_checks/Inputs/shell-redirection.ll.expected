; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py
;; Check that we can parse RUN lines that have shell redirections
; RUN: llc -mtriple=x86_64 %s -o - 2>/dev/null | FileCheck %s --check-prefix=I32
; RUN: llc < %s -mtriple=x86_64 2>&1 | FileCheck %s --check-prefix=I32
; RUN: sed 's/i32/i64/g' %s | llc -mtriple=x86_64 2>&1 | FileCheck %s --check-prefix=I64

define i32 @add(i32 %X, i32 %Y) {
; I32-LABEL: add:
; I32:       # %bb.0:
; I32-NEXT:    # kill: def $esi killed $esi def $rsi
; I32-NEXT:    # kill: def $edi killed $edi def $rdi
; I32-NEXT:    leal (%rdi,%rsi), %eax
; I32-NEXT:    retq
;
; I64-LABEL: add:
; I64:       # %bb.0:
; I64-NEXT:    leaq (%rdi,%rsi), %rax
; I64-NEXT:    retq
  %Q = add i32 %X, %Y
  ret i32 %Q
}
